### PR TITLE
[Windows] Add console script entry point

### PR DESCRIPTION
**Pull request by seanyen**
Add console script entry points for platforms (e.g. Windows) not supporting shebang to run the Scripts.

### CODE REVIEW

File: src/rosdep2/main.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def _rosdep_main(args):
         installer_keys = get_default_installer()[1]
         version_strings = []
         for key in installer_keys:
-            if key is 'source':
+            if key == 'source':
```
Comment (seanyen-msft) [inline src/rosdep2/main.py]: Flake8 fix for `F632 use ==/!= to compare str, bytes, and int literals`.

**Comment by seanyen** [comment]
@nuclearsandwich @wjwwood Ping! it would be nice I can get some feedbacks on this change, and let me know anything needed to be changed.

And btw, I think this can address this issue: https://github.com/ros-infrastructure/rosdep/issues/645

**Comment by nuclearsandwich** [comment]
I will have to check, as I don't know off the top of my head, how far back the entry_points API is supported by setuptools. There may be platforms that can't make use of them yet which rosdep is still released for.

**Comment by seanyen** [comment]
Run a quick check, `console_scripts` was there in setuptools even when it is still in a pre-release state (0.6a2, which was dated back to 2005.) Given how long it has been exercised by other projects\platforms, presumably it should be matured enough not to give people suprise. Hope that gives some more confident from other perspectives. 😃  

https://setuptools.readthedocs.io/en/latest/history.html#a2

**Comment by seanyen** [comment]
@nuclearsandwich Just checking back any feedbacks on this change?

**Comment by seanyen** [comment]
@sloretz Was it something we could consider to merge?

**Comment by seanyen** [comment]
@sloretz @dirk-thomas Happy new year. This is ready for review and merge. Hope we can move `rosdep` `Windows` support forward.

NOTE: The CI failure seems unrelated to the change and gets repro on `master` too.

**Comment by seanyen** [comment]
@wjwwood Ping! This is ready to review and merge. Let me know any additional tests to run.

**Comment by wjwwood** [comment]
My latest comment said it needed to be tested on Linux to ensure it works as it does without the patch. Has that been done?

**Comment by seanyen** [comment]
> My latest comment said it needed to be tested on Linux to ensure it works as it does without the patch. Has that been done?

@wjwwood Not sure how extensive the matrix I should validate, but I ran it on containers (python3\debian10, python2.7\debain10, and python2.7/alpine3.10) and Ubuntu bionic WSL. I can run `rosdep init` and `rosdep update` and see it is working with this patch.

**Comment by seanyen** [comment]
@tfoote Thanks for catching this. I updated the check.

**Comment by nuclearsandwich** [comment]
This change passed CI when rebased as #749.

**Pull request merged**